common/maplib/map.mjs

```diff
@@ -192,6 +211,9 @@ class _SingleGameMap {
           dynamic: false
         });
       }
+      if(cellSet.name == "SpawnPoint"){
```

Comment:
Consider creating a hash table style map from cellSet name to the cellSet in this.gameMap.cellSet.
This would eliminate the need to store the SpawnPoint cellset's .cells in this.spawnpointCellSet. You can directly grab the cellSet's .cell in getSpawnPoint() through the map.